modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy

```diff
@@ -449,7 +449,7 @@ class BashWrapperBuilder {
                 }
                 return path
             }
-            catch (FileSystemException | SocketException | RuntimeException e) {
+            catch (FileSystemException | IOException | RuntimeException | SocketException e) {
```

Comment:
The problem is IOException is very generic, even if the path is wrong it will enter in a retry loop

Reply:
Oof, then we need to handle this at the level of the GCS filesystem. We might need to implement a wrapper around the NIO filesystem provided by the google SDK...

Reply:
Fortunately I have done most of that in [this branch](https://github.com/nextflow-io/nextflow/tree/3372-google-nio-filesystem), if we want to go down that route

Reply:
A simple hack could be to check the class name (or simple name) as a string

Reply:
@bentsherman any clue whats the google base class name for those errors?

Reply:
it directly extends IOException so your solution is the best we can do without adding google deps